R/content-tools.R

```diff
@@ -77,13 +77,26 @@ new_tool_result <- function(request, result = NULL, error = NULL) {
 
 # Also need to handle edge cases: https://platform.openai.com/docs/guides/function-calling/edge-cases
 invoke_tool <- function(request) {
-  if (is.null(request@tool)) {
+  tool <- request@tool
+
+  if (is.null(tool)) {
     return(new_tool_result(request, error = "Unknown tool"))
   }
 
+  args <- request@arguments
+  if (tool@convert) {
+    extra_args <- setdiff(names(args), names(tool@arguments@properties))
+    if (length(extra_args) > 0) {
+      e <- catch_cnd(cli::cli_abort("Unused argument{?s}: {extra_args}"))
+      return(new_tool_result(request, error = e))
+    }
+
+    args <- convert_from_type(args, tool@arguments)
+  }
```

Comment:
We need to do this for `invoke_tool_async()` too, right? Given that, it'd probably be best to factor this out into a helper function

Reply:
I'm working on this now

Reply:
Done in 9c6e24eaedf5c48434d5944f7292257b36d2f881